simulate/simulate.go

```diff
@@ -501,3 +513,33 @@ func RemoveAncestors(filename string, tree *expandedTree.ETree, outputFilename s
 	outFile = outputFilename
 	fasta.Write(outFile, newFastas)
 }
+
+// seqContainsDels checks whether a deletion is present in slice of bases.
+func seqContainsDels(seq []dna.Base, del dna.Base) bool {
+	for _, base := range seq {
+		if base == del {
+			return true
+		}
+	}
+	return false
+}
+
+// codonContainsDels checks whether a deletion is present in codon.
+func codonContainsDels(codon CodonExt, del dna.Base) bool {
+	for _, baseExt := range codon.Seq {
+		if baseExt.Base == del {
+			return true
+		}
+	}
+	return false
+}
+
+// deleteBase replaces a single random base from provided sequence with deletion symbol(hyphen)
+func deleteBase(seq []dna.Base) []dna.Base {
+	delPos := rand.Intn(len(seq) - 3) // select random sequence position for deletion; avoid stop codon
```

Comment:
This may be a good place to use numbers.RandIntInRange.  If not, you can use 3 + rand.Intn(len(seq)-6) to avoid both the start and stop codons

Reply:
Thanks for pointing me to this function, was not aware of its existence.